homeassistant/components/tplink/switch.py

```diff
@@ -93,6 +93,9 @@ def __init__(
         super().__init__(device, coordinator)
         # For backwards compat with pyHS100
         self._attr_unique_id = legacy_device_id(device)
+        # Define names for individual sockets
+        if device.is_strip_socket:
+            self._attr_name = device.alias
```

Comment:
I'm not sure this will always match the naming scheme

Reply:
Maybe a function to reformat the `alias` ?

Reply:
So it's not clear to me which exact rule this would break. Should the first letter just be uncapitalized, or should the name include also the fact that it's an individual socket?
So some variant of:
* KP303 socket named kitchen
* KP303 socket kitchen
* KP303 kitchen (socket)
?

Reply:
https://developers.home-assistant.io/blog/2022/07/10/entity_naming?_highlight=entity&_highlight=name


`Device` `Entity lower lower lower`

I think something like
```python
def format_alias(alias: str) -> str:
    """Format an alias."""
    return " ".join(
        part.title() if idx == 0 else part.lower()
        for idx, part in enumerate(alias.split(" "))
    )
```

Reply:
While writing tests for the formating, it struck me that it's a bit more complicated than that if we want to follow the "unless it's a word that represents a brand/name/abbreviation of course" part of the rule. "iPhone Charger" would transform to "Iphone charger", just to give an example.

Reply:
I think it might be a bit of an edge case for a user to name their device after a brand unless maybe they put tplink in it?

Reply:
Fair enough, the names are configurable anyway. In its current form the entity names would be like this which is probably not what's expected, so I think I'll have to take another look into this later:
* `switch.iphone_charger_iphone_charger`
* `switch.iphone_charger_todays_consumption` 
* ..